rocksdb/rocksdb.nim

```diff
@@ -0,0 +1,251 @@
+# Nim-RocksDB
+# Copyright 2024 Status Research & Development GmbH
+# Licensed under either of
+#
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE) or http://www.apache.org/licenses/LICENSE-2.0)
+#  * GPL license, version 2.0, ([LICENSE-GPLv2](LICENSE-GPLv2) or https://www.gnu.org/licenses/old-licenses/gpl-2.0.en.html)
+#
+# at your option. This file may not be copied, modified, or distributed except according to those terms.
+
+{.push raises: [].}
+
+import
+  std/sequtils,
+  results,
+  ./lib/librocksdb,
+  ./options/[dbopts, readopts, writeopts],
+  ./columnfamily/[cfopts, cfdescriptor, cfhandle],
+  ./internal/[cftable, utils]
+
+export
+  results,
+  dbopts,
+  readopts,
+  writeopts,
+  cfdescriptor
+
+type
+  RocksDBResult*[T] = Result[T, string]
+
+  RocksDbPtr = ptr rocksdb_t
+
+  RocksDbRef* = ref object of RootObj
+    cPtr: RocksDbPtr
+    path: string
+    dbOpts: DbOptionsRef
+    readOpts: ReadOptionsRef
+    cfTable: ColFamilyTableRef
+
+  RocksDbReadOnlyRef* = ref object of RocksDbRef
+
+  RocksDbReadWriteRef* = ref object of RocksDbRef
+    writeOpts: WriteOptionsRef
+
+  DataProc* = proc(val: openArray[byte]) {.gcsafe, raises: [].}
```

Comment:
The purpose of using this DataProc callback is to reduce the number of copies when getting data.